### DISCUSSION TITLE
Append a new magit-log option "--perl-regexp" now working in magit-log

### DISCUSSION THREAD

**Discussion by stardiviner** [magit/magit#4950] [Support] [answered]
I try to show magit log by excluding a specific author `github-actions[bot]` from this git repository: https://github.com/immersive-translate/immersive-translate/commits/main

I found a git solution from [here](https://stackoverflow.com/questions/22968710/git-filter-log-by-group-of-authors), and [here](https://coderwall.com/p/tzdzwa/git-log-that-excludes-an-author)

I did a search in Magit issues, found a [post](https://github.com/magit/magit/issues/3511) by changing Magit transient.

Then I added new option `--perl-regexp` into `magit-log` transient with bellowing elisp code:
```elisp
(transient-append-suffix 'magit-log "-S"
    '("-P" "Toggle Perl regexp matching" "--perl-regexp"))
```
Then I press `l -P -A '^(?!(.*(github-actions|github)))'   l`, but I got `(empty)`, I confirmed the raw git command `git log --perl-regexp --author='^(?!(.*(github-actions|github)))' -n10` works fine and show commits.

```
Emacs version: GNU Emacs 30.0.50 (build 1, aarch64-apple-darwin22.4.0, NS appkit-2299.50 Version 13.3.1 (a) (Build 22E772610a)) of 2023-05-23
Magit version: Magit 20230521.2344 [>= 3.3.0.50-git], Git 2.39.2, Emacs 30.0.50, darwin
```

Anyone knows why the Magit transient log is empty, but raw git command works fine?
Hope someone can help me to fix this problem. Thanks a lot.

**Comment by tarsius** [comment, marked as answer]
When Magit runs `git`, no shell is involved, so shell quoting should not be used. You are searching for `'^(...)'`, when you meant to search for `^(...)`.

**Reply by stardiviner** [reply]
Thanks for answering.
